recipe/meta.yaml

```
@@ -274,6 +274,34 @@ outputs:
       commands:
         - pip check
 
+  - name: tensorflow-io-gcs-filesystem
+    script: build_io_gcs_fs.sh   # [not win]
+    build:
+      string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
+      string: cpu_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version == "None"]
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - {{ compiler('cuda') }}                 # [cuda_compiler_version != "None"]
+        - bazel
+        - bazel >=4.2.1  # [osx and arm64]
+      host:
+        - python
+        - pip
+        - setuptools
+        - wheel
+        - openssl
+        - {{ pin_subpackage('tensorflow-base', exact=True) }}
+      run:
+        - python
+        - {{ pin_subpackage('tensorflow-base', exact=True) }}
+    test:
+      requires:
+        - pip
+      commands:
+        - pip check
+
```

```suggestion
  - name: tensorflow-io
    script: build_io_gcs_fs.sh   # [not win]
    build:
      string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
      string: cpu_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version == "None"]
    requirements:
      build:
        - {{ compiler('c') }}
        - {{ compiler('cxx') }}
        - {{ compiler('cuda') }}                 # [cuda_compiler_version != "None"]
        - bazel
        - bazel >=4.2.1  # [osx and arm64]
      host:
        - python
        - pip
        - setuptools
        - wheel
        - openssl
        - {{ pin_subpackage('tensorflow', exact=True) }}

      run:
        - python
        - {{ pin_subpackage('tensorflow', exact=True) }}
    test:
      requires:
        - pip
      commands:
        - pip check

```

Hi @ganand1, looks you're very close. Could you try building tensorflow-io instead of just the gcs-filesystem? tensorflow-io includes tensorflow-io-gcs-filesystem. Also, I would pin against tensorflow, not the base here.